components/common/hooks/useGetLimits.ts

```diff
@@ -5,18 +5,25 @@ import {useState, useEffect, useMemo} from 'react';
 import {useDispatch, useSelector} from 'react-redux';
 
 import {Limits} from '@mattermost/types/cloud';
-import {getCloudLimits, getCloudLimitsLoaded, isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+import {getCloudLimits, getCloudLimitsLoaded, isCurrentLicenseCloud, isCloudTrialExpired} from 'mattermost-redux/selectors/entities/cloud';
 import {getCloudLimits as getCloudLimitsAction} from 'actions/cloud';
 import {useIsLoggedIn} from 'components/global_header/hooks';
 
 export default function useGetLimits(): [Limits, boolean] {
     const isCloud = useSelector(isCurrentLicenseCloud);
+    const cloudTrialExpired = useSelector(isCloudTrialExpired);
     const isLoggedIn = useIsLoggedIn();
     const cloudLimits = useSelector(getCloudLimits);
     const cloudLimitsReceived = useSelector(getCloudLimitsLoaded);
     const dispatch = useDispatch();
     const [requestedLimits, setRequestedLimits] = useState(false);
 
+    useEffect(() => {
+        if (cloudTrialExpired) {
+            setRequestedLimits(false);
+        }
+    }, [cloudTrialExpired]);
+
     useEffect(() => {
         if (isLoggedIn && isCloud && !requestedLimits && !cloudLimitsReceived) {
```

Comment:
I wonder if this might work and if we could remove the new useEffect unless we want to call the dispatch twice. that might as well be unpredictable as it would depend on which useEffect gets triggered first.
```suggestion
        if (isLoggedIn && isCloud && !requestedLimits && !cloudLimitsReceived && cloudTrailExpired) {
```